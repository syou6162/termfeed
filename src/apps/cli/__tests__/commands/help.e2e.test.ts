import { describe, it, expect } from 'vitest';
import { execSync } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const CLI_PATH = path.join(__dirname, '../../../../../dist/cli.js');

describe('help command E2E', () => {
  it('should display main help message with --help flag', () => {
    const result = execSync(`node ${CLI_PATH} --help`, { encoding: 'utf-8' });

    // メインのヘルプメッセージが表示されることを確認
    expect(result).toContain('Usage: termfeed [options] [command]');
    expect(result).toContain('A terminal-based RSS reader with Vim-like keybindings');

    // オプションが表示されることを確認
    expect(result).toContain('-V, --version');
    expect(result).toContain('-h, --help');
    expect(result).toContain('display help for command');

    // コマンドが表示されることを確認
    expect(result).toContain('add [options] <url>');
    expect(result).toContain('Add a new RSS feed');
    expect(result).toContain('rm <feedId>');
    expect(result).toContain('Remove RSS feed by ID');
    expect(result).toContain('tui');
    expect(result).toContain('Start RSS reader in TUI mode');
    expect(result).toContain('export [options] [file]');
    expect(result).toContain('Export feed subscriptions to OPML or text file');
    expect(result).toContain('import [options] <file>');
    expect(result).toContain('Import feed subscriptions from OPML or text file');
    expect(result).toContain('mcp-server');
    expect(result).toContain('Start MCP server to expose termfeed data to LLM');
    expect(result).toContain('help [command]');
    expect(result).toContain('Display help for command');
  });

  it('should display help message with help command', () => {
    const result = execSync(`node ${CLI_PATH} help`, { encoding: 'utf-8' });

    // helpコマンドでも同じメッセージが表示されることを確認
    expect(result).toContain('Usage: termfeed [options] [command]');
    expect(result).toContain('A terminal-based RSS reader with Vim-like keybindings');
  });

  it('should match add command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} add --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match rm command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} rm --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match tui command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} tui --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match export command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} export --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match import command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} import --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match mcp-server command help snapshot', () => {
    const result = execSync(`node ${CLI_PATH} mcp-server --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });

  it('should match help output snapshot', () => {
    const result = execSync(`node ${CLI_PATH} --help`, { encoding: 'utf-8' });
    expect(result).toMatchSnapshot();
  });
});
