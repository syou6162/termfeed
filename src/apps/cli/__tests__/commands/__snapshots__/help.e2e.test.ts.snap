// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`help command E2E > should match add command help snapshot 1`] = `
"Usage: termfeed add [options] <url>

Add a new RSS feed

Arguments:
  url                  RSS feed URL to add

Options:
  -t, --title <title>  Custom title for the feed
  -h, --help           display help for command
"
`;

exports[`help command E2E > should match export command help snapshot 1`] = `
"Usage: termfeed export [options] [file]

Export feed subscriptions to OPML or text file

Arguments:
  file                   output file path (default: subscriptions.opml)

Options:
  -f, --format <format>  Export format (opml or text) (default: "auto")
  -h, --help             display help for command
"
`;

exports[`help command E2E > should match help output snapshot 1`] = `
"Usage: termfeed [options] [command]

A terminal-based RSS reader with Vim-like keybindings

Options:
  -V, --version            output the version number
  -h, --help               display help for command

Commands:
  add [options] <url>      Add a new RSS feed
  rm <feedId>              Remove RSS feed by ID
  tui                      Start RSS reader in TUI mode
  export [options] [file]  Export feed subscriptions to OPML or text file
  import [options] <file>  Import feed subscriptions from OPML or text file
  mcp-server               Start MCP server to expose termfeed data to LLM
                           agents
  help [command]           Display help for command
"
`;

exports[`help command E2E > should match import command help snapshot 1`] = `
"Usage: termfeed import [options] <file>

Import feed subscriptions from OPML or text file

Arguments:
  file                   file to import

Options:
  -f, --format <format>  Import format (opml or text) (default: "auto")
  -h, --help             display help for command
"
`;

exports[`help command E2E > should match mcp-server command help snapshot 1`] = `
"Usage: termfeed mcp-server [options]

Start MCP server to expose termfeed data to LLM agents

Options:
  -h, --help  display help for command
"
`;

exports[`help command E2E > should match rm command help snapshot 1`] = `
"Usage: termfeed rm [options] <feedId>

Remove RSS feed by ID

Arguments:
  feedId      Feed ID to remove

Options:
  -h, --help  display help for command
"
`;

exports[`help command E2E > should match tui command help snapshot 1`] = `
"Usage: termfeed tui [options]

Start RSS reader in TUI mode

Options:
  -h, --help  display help for command
"
`;
